packages/assets-controllers/src/NftController.ts

```diff
@@ -1143,15 +1177,21 @@ export class NftController extends BaseControllerV1<NftConfig, NftState> {
 
   // temporary method to get the correct chainId until we remove chainId from the config & the chainId arg from the detection logic
   // Just a helper method to prefer the networkClient chainId first then the chainId argument and then finally the config chainId
-  private getCorrectChainId({
+  #getCorrectChainId({
     networkClientId,
   }: {
     networkClientId?: NetworkClientId;
   }) {
     if (networkClientId) {
-      return this.getNetworkClientById(networkClientId).configuration.chainId;
+      const {
+        configuration: { chainId },
+      } = this.messagingSystem.call(
+        'NetworkController:getNetworkClientById',
+        networkClientId,
+      );
+      return chainId;
     }
-    return this.config.chainId;
+    return this.#config.chainId;
   }
 
```

Comment:
It looks like we have a better solution for this in the `TokenDetectionController`'s `#getCorrectChainIdAndNetworkClientId` method. Maybe we could extract that into a utility method and use it both here and in `TokenDetectionController`?

@mcmire Do you think this might be a good idea?

Reply:
I'm okay with copying what we did for other controllers as long as it's easy. However, there is a fair amount of inconsistency in the controllers that have been multichain-ized — every one of them works slightly differently — and my worry is that we may have to change a bunch of stuff about this controller to fit what another controller is doing. We definitely need to spend some time at some point on unifying them, but I'm not sure we should make any new abstractions yet. So I'd vote to keep it simple and just make whatever changes are necessary to convert from callbacks to the messenger object. I haven't looked at the other parts of this PR yet so I may have a different opinion, but that's what I think so far at least :)

Reply:
ce can maybe refactor this in a future PR and align the usage across controllers ?

Reply:
Yes, as long as we're getting the accurate `chainId` with this method, we should be able to keep it as is for now.